src/main/java/org/jabref/gui/util/BackgroundTask.java

```diff
@@ -118,6 +129,7 @@ public BackgroundProgress getProgress() {
      * Sets the {@link Consumer} that is invoked after the task is successfully finished.
      * The consumer always runs on the JavaFX thread.
      */
+    @Deprecated
```

Comment:
why did you deprecated the onSuccess and onFailure method?

Reply:
This functionality is now handled by the `CompletableFuture` object. See [here](https://www.logicbig.com/tutorials/core-java-tutorial/java-multi-threading/completion-stages-exception-handling.html). I really don't like the fact that the chained task is always executed on the JavaFX thread, which potentially leads to GUI freezes. See the java doc of `Platform.runLater`:

`     * NOTE: applications should avoid flooding JavaFX with too many
     * pending Runnables. Otherwise, the application may become unresponsive.
     * Applications are encouraged to batch up multiple operations into fewer
     * runLater calls.
     * Additionally, long-running operations should be done on a background
     * thread where possible, freeing up the JavaFX Application Thread for GUI
     * operations.`

Therefore this method has to be deprecated as long as it's 'flooding' the JavaFX thread.

Reply:
These background tasks are not executed on the javafx thread! Only the onsucess/onfailure methods are, which is really nice since you usually want to display messages there which should be done on the javafx thread. Thus this functionality is absolutely not covered by the completetable future.

Reply:
The javafx Task Api is similar to the old swing worker.
Running stuff on a background thread, but having the possibility to have updates etc on the FX thread:
https://docs.oracle.com/javase/8/javafx/interoperability-tutorial/concurrency.htm